Description: Modifications to work with dh_elpa.
 * Strip down setupEmacs() function.  We only set f12 as a global key
   to start Macaulay2 inside Emacs.  This also strips (load "M2-init")
   from a user's existing .emacs-Macaulay2 file, which would cause
   errors if they haven't installed elpa-macaulay2.  We also warn the
   user if elpa-macaulay2 isn't installed.
Author: Doug Torrance <dtorrance@piedmont.edu>
Forwarded: not-needed
Last-Update: 2021-02-24

--- a/M2/Macaulay2/m2/files.m2
+++ b/M2/Macaulay2/m2/files.m2
@@ -385,12 +385,6 @@
 ///
 
 dotemacsFix0 = ///
-;; this version will give an error if M2-init.el is not found:
-(load "M2-init")
-
-;; this version will not give an error if M2-init.el is not found:
-;(load "M2-init" t)
-
 ; You may comment out the following line with an initial semicolon if you 
 ; want to use your f12 key for something else.  However, this action
 ; will be undone the next time you run setup() or setupEmacs().
@@ -472,10 +466,15 @@
 setupEmacs = method()
 setup = method()
 mungeEmacs = () -> (
-     dotemacsFix = concatenate(emacsHeader, apply(emacsfixes, (var,dir,templ) -> fix(var,dir,"",templ)), dotemacsFix0);
+     dotemacsFix = concatenate(emacsHeader, dotemacsFix0);
      supplantStringFile(dotemacsFix,"~/"|M2emacs,false);
-     mungeFile("~/"|".emacs", ";; Macaulay 2 start", ";; Macaulay 2 end", M2emacsRead )
-     )
+     ret := mungeFile("~/"|".emacs", ";; Macaulay 2 start", ";; Macaulay 2 end", M2emacsRead );
+     if run("dpkg -s elpa-macaulay2 2> /dev/null | " |
+	  "grep -q \"^Status:.* installed\"") != 0 then
+	  stderr << "--warning: you must run \"apt install elpa-macaulay2\" "
+	       << "as root in order to use" << endl
+	       <<  "  Macaulay2 with Emacs." << endl;
+     ret)
 prelim := () -> (
      promptUser = true;
      if prefixDirectory === null then error "can't determine Macaulay 2 prefix (prefixDirectory not set)";
